### PR TITLE
Disable the cache by default

### PR DESCRIPTION
**Pull request by lbargaoanu**
Breaking change...

### CODE REVIEW

File: src/AutoMapper/IMappingOperationOptions.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ public interface IMappingOperationOptions
         /// <summary>
         /// Disable the cache used to re-use destination instances based on equality
         /// </summary>
-        bool DisableCache { get; set; }
+        bool PreserveReferences { get; set; }
```
Comment (jbogard) [inline src/AutoMapper/IMappingOperationOptions.cs]: Let's move this up to the TypeMap config level. Disabling is per operation but enabling is per map.

**Comment by lbargaoanu** [comment]
MaxDepth enables the cache.

**Comment by jbogard** [comment]
I think I broke your merge :(

**Comment by lbargaoanu** [comment]
Rebased.

**Pull request merged**